code/modules/mob/living/silicon/ai/freelook/chunk.dm

```diff
@@ -9,40 +7,76 @@
 	var/list/obscuredTurfs = list()
 	var/list/visibleTurfs = list()
 	var/list/obscured = list()
-	var/list/cameras = list()
+	var/list/active_cameras = list()
+	var/list/inactive_cameras = list()
 	var/list/turfs = list()
 	var/list/seenby = list()
-	var/visible = FALSE
 	var/changed = FALSE
 	var/updating = FALSE
 	var/x = 0
 	var/y = 0
 	var/z = 0
 
-// Add an AI eye to the chunk, then update if changed.
+/datum/camerachunk/proc/add_camera(obj/machinery/camera/cam)
+	if(active_cameras[cam] || inactive_cameras[cam])
+		return
+	// Register all even though it is active/inactive. Won't get called incorrectly
+	RegisterSignal(cam, COMSIG_CAMERA_OFF, PROC_REF(deactivate_camera))
+	RegisterSignal(cam, COMSIG_CAMERA_ON, PROC_REF(activate_camera))
+	RegisterSignal(cam, COMSIG_PARENT_QDELETING, PROC_REF(remove_camera))
+	RegisterSignal(cam, COMSIG_CAMERA_MOVED, PROC_REF(camera_moved))
+	if(cam.can_use())
+		active_cameras[cam] = cam
+	else
+		inactive_cameras[cam] = cam
+
+/datum/camerachunk/proc/camera_moved(obj/machinery/camera/cam, atom/old_loc)
+	var/turf/T = get_turf(cam)
+
+	// Falls outside of the chunk view distance
+	if(T.x + CAMERA_VIEW_DISTANCE < x || T.x - CAMERA_VIEW_DISTANCE >= x + CAMERA_CHUNK_SIZE || T.y + CAMERA_VIEW_DISTANCE < y || T.y - CAMERA_VIEW_DISTANCE >= y + CAMERA_CHUNK_SIZE)
```

Comment:
are there any checks here for cams travelling between z levels? Would be worth including